src/WorldMap.cpp

```diff
@@ -1,33 +1,228 @@
+	assert(textureLoaded);
+}
+
+WorldMap::TileSheet::~TileSheet()
+{
+	delete m_texture;
+	m_texture = nullptr;
+}
+
+WorldMap::TileSheet::TileSheet(const TileSheet& orig)
+	: m_tileSheetDetails(orig.m_tileSheetDetails),
+	m_filePath(orig.m_filePath)
+{
+	m_texture = new sf::Texture(*orig.m_texture);
+}
+
+const std::string& WorldMap::TileSheet::getFilePath() const
+{
+	return m_filePath;
+}
+
+const WorldMap::TileSheetDetails& WorldMap::TileSheet::getDetails() const
+{
+	return m_tileSheetDetails;
+}
+
+const sf::Texture& WorldMap::TileSheet::getTexture() const
+{
+	assert(m_texture);
+	return *m_texture;
+}
+
+sf::IntRect WorldMap::TileSheet::getTileLocationByID(int ID) const
+{
+	//Scan through every tile on tilesheet to find appropriate tile by ID
+	int col = 0, row = 0;
+	const int firstTileID = 0;
+	const int endTileID = m_tileSheetDetails.m_rows * m_tileSheetDetails.m_columns;
+
+	for (int i = firstTileID; i < endTileID; ++i)
+	{
+		//Appropriate tile has been found
+		if (i == ID)
+		{
+			break;
+		}
+
+		//If tile not found, move onto the next tile in the row
+		++col;
+
+		//If reaching end of row, move onto the next
+		if (col == m_tileSheetDetails.m_columns)
+		{
+			col = 0;
+			++row;
+		}
+	}
+
+	//Get location on the tile sheet
+	const int tileSize = m_tileSheetDetails.m_tileSize;
+	const int margin = m_tileSheetDetails.m_margin;
+	const int spacing = m_tileSheetDetails.m_spacing;
+
+	return sf::IntRect((margin + ((spacing + tileSize) * col)),
+		(margin + ((spacing + tileSize) * row)),
+		tileSize, tileSize);
+}
+
+sf::IntRect WorldMap::TileSheet::getTileLocationByPosition(const sf::IntRect& position) const
+{
+	const int tileSize = m_tileSheetDetails.m_tileSize;
+	return sf::IntRect(position.left * tileSize, position.top * tileSize, position.width * tileSize, position.height * tileSize);
+}
+
+//TileLayer.cpp
+WorldMap::TileLayer::Tile::Tile(const sf::Vector2i& position, const TileSheet& tileSheet, int tileID)
+	: m_position(position),
+	m_tileID(tileID),
+	m_tileSheet(tileSheet)
+{
+	m_sprite.setTexture(m_tileSheet.getTexture());
+	m_sprite.setTextureRect(m_tileSheet.getTileLocationByID(m_tileID));
+	const int tileSize = m_tileSheet.getDetails().m_tileSize;
+	m_sprite.setPosition(sf::Vector2f(m_position.x * tileSize, m_position.y * tileSize));
+}
+
+WorldMap::TileLayer::Tile::Tile(const Tile& orig)
+	: m_position(orig.m_position),
+	m_tileID(orig.m_tileID),
+	m_tileSheet(orig.m_tileSheet)
+{
+	m_sprite.setTexture(m_tileSheet.getTexture());
+	m_sprite.setTextureRect(m_tileSheet.getTileLocationByID(m_tileID));
+	const int tileSize = m_tileSheet.getDetails().m_tileSize;
+	m_sprite.setPosition(sf::Vector2f(m_position.x * tileSize, m_position.y * tileSize));
+}
+
+WorldMap::TileLayer::TileLayer(const std::vector<std::vector<int>>& tileMapData, const sf::Vector2i& mapSize, const TileSheet& tileSheet, std::string&& name)
+	: m_name(std::move(name)),
+	m_tileMap()
+{
+	loadInTileMap(tileMapData, mapSize, tileSheet);
```

Comment:
Hm.  Personally, I'm not a huge fan of single-use functions like this - you could just move the body of loadInTileMap into this constructor.  It would make it more obvious that the 'if (!m_tileMap.empty())' check is unnecessary.  But, this is a minor stylistic preference so I'll leave this one up to you.